### DISCUSSION TITLE
Don't show Navigator->OPEN EDITORS by default

### DISCUSSION THREAD

**Discussion by wuestkamp** [eclipse-theia/theia#10935] [Q&A] [answered]
How can I hide the OPEN EDITORS section in the Navigator by default?
Is this possible with a custom extension?

It looks like its added here:
https://github.com/eclipse-theia/theia/blob/master/packages/navigator/src/browser/navigator-widget-factory.ts#L71

But how can I override this?

Thanks,
Kim

**Comment by msujew** [comment, marked as answer]
Hi @wuestkamp,

this is a [known issue](https://github.com/eclipse-theia/theia/issues/10928), it shouldn't actually be expanded by default. Luckily, someone already provided [a PR](https://github.com/eclipse-theia/theia/pull/10930) to fix this. If you want to remove the view completely, you can simply override the `NavigatorWidgetFactory` and modify the `createWidget` so that it doesn't add the widget in the first place.

**Reply by wuestkamp** [reply]
It's possible to override using an extension? Because its not an interface like with for example MenuContribution...
Are the any extension examples to look at that do this kind of thing?

Thanks!

**Reply by msujew** [reply]
Ah, yeah we don't have any samples for that use case. However, you can accomplish it like this:

```ts
@injectable()
export class CustomNavigatorWidgetFactory extends NavigatorWidgetFactory {

    async createWidget(): Promise<ViewContainer> {
        const viewContainer = this.viewContainerFactory({
            id: EXPLORER_VIEW_CONTAINER_ID,
            progressLocationId: 'explorer'
        });
        viewContainer.setTitleOptions(EXPLORER_VIEW_CONTAINER_TITLE_OPTIONS);
        // const openEditorsWidget = await this.widgetManager.getOrCreateWidget(OpenEditorsWidget.ID);
        const navigatorWidget = await this.widgetManager.getOrCreateWidget(FILE_NAVIGATOR_ID);
        // viewContainer.addWidget(openEditorsWidget, this.openEditorsWidgetOptions);
        viewContainer.addWidget(navigatorWidget, this.fileNavigatorWidgetOptions);
        return viewContainer;
    }
}
```

And then rebind it:

```ts
export default new ContainerModule((bind, unbind, isBound, rebind) => {
    rebind(NavigatorWidgetFactory).to(CustomNavigatorWidgetFactory).inSingletonScope();
}
```

**Reply by wuestkamp** [reply]
Thanks a lot! Works great!
